crates/runtime/src/dataconnector/odbc.rs

```
@@ -47,6 +53,20 @@ where
     odbc_factory: ODBCTableFactory<'a>,
 }
 
+fn match_driver_to_dialect(driver: &str) -> Option<Arc<dyn Dialect + Send + Sync>> {
+    match driver {
+        _ if driver.contains("mysql") => Some(Arc::new(MySqlDialect {})), // odbcinst.ini profile name
+        _ if driver.contains("libmyodbc") => Some(Arc::new(MySqlDialect {})), // library filename
+        _ if driver.contains("postgres") => Some(Arc::new(PostgreSqlDialect {})), // odbcinst.ini profile name
+        _ if driver.contains("psqlodbc") => Some(Arc::new(PostgreSqlDialect {})), // library filename
+        _ if driver.contains("sqlite") => Some(Arc::new(SqliteDialect {})), // profile and library name
+        _ => {
+            tracing::debug!("No dialect detected for driver: {}", driver);
```

Might be better to return an error and handle it in the caller below